### ISSUE TITLE
[bug] Pipeline UI couldn't load mlpipline-ui-metadata (v1) with Minio gateway in Multi-user install on GCP

### ISSUE THREAD

**Issue by hainp** [kubeflow/pipelines#6750]
### What steps did you take
I ran a simple 1 step pipeline, which aimed to display a webpage in visualization UI (the Output).

### What happened:

In a KF cluster with MinIO gateway (GCS is the actual storage), Multi-users install (on GCP), the Visualization UI couldn't display the webpage.
[Screenshot](https://imgur.com/5rC0YDg)

### What did you expect to happen:

In a KF cluster with in-cluster MinIO setup (a PD is the actual storage), Multi-users install (on GCP), the Visualization UI could display the webpage
[Screenshot](https://imgur.com/8IZn43o)

### Environment:

Platform: GCP, IAP-ingress enabled, CloudSQL, MinIO gateway (with a GCS bucket)

*  How do you deploy Kubeflow Pipelines (KFP)? 
A custom from kubeflow/gcp-blueprints, because of my company policies, CloudSQL/MySQL is connected via a private connection (VPC) instead of using cloudsql proxy. I also can't use Cluster management.
*  KFP version: 1.7.0
*  KFP SDK version: 
kfp                            1.8.4
kfp-pipeline-spec      0.1.12
kfp-server-api           1.7.0

### Anything else you would like to add:
The Visualization UI looks like loading an artifact with namespace
`https://{host}/pipeline/artifacts/minio/om-ml-pipeline-np/artifacts/viz-sample-rtdtl/2021/10/17/viz-sample-rtdtl-2393626054/mlpipeline-ui-metadata.tgz?namespace=om`

Clicking to the link I should get:
`Failed to get object in bucket om-ml-pipeline-np at path artifacts/viz-sample-rtdtl/2021/10/17/viz-sample-rtdtl-2393626054/mlpipeline-ui-metadata.tgz: S3Error: The request signature we calculated does not match the signature you provided. Check your key and signing method.`

Remove the `?namespace=om` in the link then I can download the artifact.

I've already follow https://github.com/kubeflow/pipelines/issues/4649 , but v2 is still in Beta and lacks some features I need in v1, so I have to stick with v1.

### Labels

/area frontend
/area backend
/area sdk

---

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by zijianjoy** [comment]
If you are using V2 compatible mode, can you trying using the new approach to visualize your metrics instead of using `mlpipeline-ui-metadata`? https://www.kubeflow.org/docs/components/pipelines/sdk/output-viewer/#v2-sdk-use-sdk-visualization-apis

**Comment by hainp** [comment]
1. No , I don't use __V2 compatible mode__ ; it still misses some features I need (eg: resource quota request / limit, packaging components..).

I add here the simple code if it could provide any insight.
```
import kfp
import kfp.dsl as dsl
import kfp.components as comp

def web_app(
    mlpipeline_ui_metadata_path: kfp.components.OutputPath(),
    ):

  import json

  metadata = {
    'outputs': [{
      'type': 'web-app',
      'storage': 'inline',
      'source': '<h1>Hello, World!</h1>',
    }]
  }

  with open(mlpipeline_ui_metadata_path, 'w') as metadata_file:
    json.dump(metadata, metadata_file)


create_web_app_task = comp.create_component_from_func(
  func=web_app,
  output_component_file="web_app_component.yaml",
  base_image="python:3.7"
)


@dsl.pipeline(name="viz_sample")
def the_pipeline():
  web_app_task = create_web_app_task()


def main():
  kfp.compiler.Compiler().compile(pipeline_func=the_pipeline, package_path="pipeline.yaml")


if __name__ == "__main__":
  main()
```

2. In __V2 compatible mode___, if `package_root` is set to (directly) `gs://{bucket}`, visualization UI works fine. if `package_root` is set to MinIO gateway `minio://{bucket}` I have the same error as posted.

**Comment by zijianjoy** [comment]
As we released KFP 1.8.1, it should include the mentioned fix. Feel free to reopen if the issue persists.
